### PR TITLE
Hero buttons

### LINKED ISSUE #5072: Hero Buttons component
Tracked by
* #5000 

## Hero Buttons
* first section are buttons of the collection, mainly socials and website.
* Then there is share button and more options - they are divided to make clear that they are not associated with the specific collection 
    * Share button is pretty clear, more options button is in case visitor - report button, and in case of owner its going to be a customize and other functionalities we already offer. 
    
image

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #5072 
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image
image
image
image

### CODE REVIEW

**Comment by daiagi** [comment]
continue discussion from issue #5072 

> the first set of icons, on the left, are collection specific - the user input the links when creating and when customizing the collection page (tbd design from me)

did not see this sort of data when query for CollectionEntity, where might i find it? is it not yet implemented? @roiLeo 
@exezbcz  should the buttons be disabled if the user did not provide these links?

> share button works similarly to gallery item, that means that once you click, it opens dropdown with share options - copy link or share on twitter
@exezbcz  I'm not sure how the dropdown is designed, for now I've copied the design from gallery item

> more button also opens a dropdown with options depending whether you are the owner or visitor
    in case of visitor, there is simple option to report
    on the other hand, owner can manage his collection through this, so options like customize, delete etc are there, this is also going to be specified once I come up with customization modal or something like that


here also i'm not sure how it is designed and are these functions/pages implemented, for now i've just put simple placeholders in the dropdown

**Comment by daiagi** [comment]
tested on /bsx/collection/3030053998?page=1&redesign=true

**Comment by exezbcz** [comment]
@daiagi hey hey 
the buttons won't be there if the owner dont fill it
- i think we dont have the data yet, correct me if yes :D 

drop down same as gallery 
image
share as well

**Comment by daiagi** [comment]
thanks @exezbcz 
> drop down same as gallery

mind pointing me to the mentioned gallery page?
all I'm finding is the old design

**Comment by roiLeo** [comment]
> did not see this sort of data when query for CollectionEntity, where might i find it? is it not yet implemented? @roiLeo @exezbcz should the buttons be disabled if the user did not provide these links?

Maybe use user identity data, hide btn when link doesn't exist

> here also i'm not sure how it is designed and are these functions/pages implemented, for now i've just put simple placeholders in the dropdown

Not ready yet ref #4983, IMO we should hide it for now (disabled is an option)

**Comment by daiagi** [comment]
> Maybe use user identity data, hide btn when link doesn't exist

user identity data?! aren't these links collection specific?

**Comment by roiLeo** [comment]
> user identity data?! aren't these links collection specific?

Nope. It's not possible (yet) to set "links" attached to collectionEntity.

File: components/collection/HeroButtons.vue [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      <NeoButton icon="discord" icon-pack="fab" class="square-32" />
+
+      <div class="vertical-seperator mx-4" />
+
+      <NeoDropdown append-to-body>
+        <NeoButton icon="share-alt" class="square-32 mr-3" />
+        <template #items>
+          <NeoDropdownItem
+            v-clipboard:copy="realworldFullPath"
+            @click.native="toast($i18n.t('toast.urlCopy'))">
```
[the remark this answers is not in the document]
Comment (daiagi) [reply components/collection/HeroButtons.vue]: very good. Thanks

File: components/collection/HeroButtons.vue [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import { NeoButton, NeoDropdown, NeoDropdownItem } from '@kodadot1/brick'
+import { isOwner as checkOwner } from '@/utils/account'
+import {
+  CollectionEntityMinimal,
+  useCollectionMinimal,
+} from './utils/useCollectionDetails'
+const route = useRoute()
+const { accountId } = useAuth()
+const { $i18n, $buefy } = useNuxtApp()
+const collection = ref<CollectionEntityMinimal>()
```
[the remark this answers is not in the document]
Comment (daiagi) [reply components/collection/HeroButtons.vue]: correct. Thank you

File: layouts/explore-layout.vue [hunk 3]
```diff
@@ -38,6 +42,7 @@
 <script lang="ts" setup>
 import ExploreTabsFilterSort from '@/components/explore/ExploreIndex.vue'
 import MobileFilter from '@/components/explore/MobileFilter.vue'
+import HeroButtons from '~~/components/collection/HeroButtons.vue'
```
[the remark this answers is not in the document]
Comment (daiagi) [reply layouts/explore-layout.vue]: nice catch

**Review by Jarsen136 — APPROVED** [review]

**Pull request merged**